third_party/move/move-compiler-v2/src/function_checker.rs

```diff
@@ -89,156 +93,175 @@ pub fn check_access_and_use(env: &mut GlobalEnv) {
                     },
                 };
 
-                // Check that inline functions being called are accessible
-                let optional_def = caller_func.get_def();
-                if let Some(def) = optional_def {
+                // Check that functions being called are accessible.
+                if let Some(def) = caller_func.get_def() {
                     let callees_with_sites = def.called_funs_with_callsites();
-                    for (callee, sites) in callees_with_sites {
-                        let callee_env = env.get_function(callee);
-                        // check visibility if not in the same module
-                        let callee_is_inline = callee_env.is_inline();
-                        let callee_is_accessible = if callee_env.module_env.get_id()
-                            == caller_module_id
-                        {
+                    for (callee, sites) in &callees_with_sites {
+                        let callee_env = env.get_function(*callee);
+                        // Check visibility.
+
+                        // Same module is always visible
+                        let same_module = callee_env.module_env.get_id() == caller_module_id;
+                        let call_involves_inline_function =
+                            callee_env.is_inline() || caller_func.is_inline();
+
+                        // SKIP check if same_module or
+                        // if before inlining and the call doesn't involve inline function
+                        // OR if after inlining and call *does* involve an inline function
```

Comment:
After inlining, there should not be any inline function left, so this comment is a bit confusing. Is it possible that we are at "after inlining", and the call *does* involve an inline function?